sylt-compiler/src/intermediate.rs

```diff
@@ -7,13 +7,12 @@ use sylt_parser::{
 
 use crate::{typechecker::TypeChecker, NamespaceID};
 
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq)]
 pub struct Var(pub usize);
 
-impl Display for Var {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let Var(n) = self;
-        write!(f, "V{}", n)
+impl Var {
+    pub fn format(self) -> String {
+        format!("V{}", self.0)
```

Comment:
This should probably still be the Display trait. `to_string` could be used instead since it gets automatically implemented.